providers/gitlab.go

```diff
@@ -206,15 +228,67 @@ func (p *GitLabProvider) EnrichSessionState(ctx context.Context, s *sessions.Ses
 		return fmt.Errorf("user email is not verified")
 	}
 
-	// Check group membership
-	// TODO (@NickMeves) - Refactor to Authorize
-	err = p.verifyGroupMembership(userInfo)
-	if err != nil {
-		return fmt.Errorf("group membership check failed: %v", err)
-	}
+	p.addGroupMembershipToState(ctx, s)
+
+	p.addProjectMembershipToState(ctx, s)
 
 	s.User = userInfo.Username
 	s.Email = userInfo.Email
 
 	return nil
+
+}
+
+// addProjectMembership adds projects into session.Groups
+func (p *GitLabProvider) addGroupMembershipToState(ctx context.Context, s *sessions.SessionState) {
```

Comment:
NIT: We either call it SessionState or in a lot of places shorten it to just Session. These function names might be cleaner as something like addGroupsToSession

On first cursory glance, I was confused at first since we use the term State in parts of the request Context for something unrelated.

Reply:
https://github.com/factorysh/oauth2-proxy/commit/c2208668dc8cff916a5a1c811b99391612ac88b9